staging/src/k8s.io/apiserver/pkg/util/proxy/metrics/metrics.go

```diff
@@ -43,19 +43,37 @@ var (
 		},
 		[]string{statuscode},
 	)
+	// streamTunnelRequestsTotal counts the number of requests that were handled by
+	// the StreamTunnelProxy (PortForward subprotocol).
+	streamTunnelRequestsTotal = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Subsystem:      subsystem,
+			Name:           "stream_tunnel_requests_total",
+			Help:           "Total number of requests that were handled by the StreamTunnelProxy, which processes streaming PortForward/V2",
+			StabilityLevel: metrics.ALPHA,
+		},
+		[]string{statuscode},
```

Comment:
this only considers the backend status code, right?

Reply:
Not necessarily. 

1. Within the `TunnelingHandler` there is one `BadRequest` response if the request does not contain subprotocols. In this case we return the 400-level response (and increment the metric) but do not consider either the backend/upstream SPDY upgrade status code or the WebSocket upgrade.

2. Within the `InitializeWrite`, we consider the backend/upstream SPDY upgrade response. Only after we determine the SPDY upgrade succeeds, do we attempt the WebSocket upgrade. So the successful status code represents both the successful SPDY upgrade and the successful WebSocket upgrade. It is also possible in the situation where the SPDY upgrade was successful, but the WebSocket upgrade was not (see your next comment), where the metric is incremented with a 500-level error response.

Please let me know what you think.

Reply:
I'm thinking more about an end user or admin or SRE and how it will interpret this metrics, in case this metrics goes up ... where should they start looking , apiserver, kubelet, containerd?

Reply:
I was thinking that if the non-successful metrics rise, investigators would initially look at the error statements in the apiserver logs. I think they would have to look at these logs no matter what. Please let me know if we should separate the SPDY from WebSocket status codes. The one place where the SPDY vs. WebSocket status code might be indeterminate is the `TunnelingHandler.ServeHTTP` returning a `BadRequest` status code.

Reply:
Nah, too complex to be able to express with metrics 

/Lgtm

Thanks for the explanation